### PR TITLE
fix: improve adapter and model for supports any javascript platforms

### PR DESCRIPTION
**Pull request by nodece**
issues: #257 

BREAKING CHANGE:
- remove `enforcer.loadModel()`
- `enforcer.initWithFile()` has been deprecated, use `initWithModelAndAdapter()` instead
- `enforcer.initWithString()` has been deprecated, use `initWithModelAndAdapter()` instead
- `enforcer.initWithAdapter()` has been deprecated, use `initWithModelAndAdapter()` instead
- `newModel()` has been deprecated, use `new Model()` instead
- `Config.newConfig()` has been deprecated, use `Config.newConfigFromText()` instead
- remove `readFile` and `writeFile` from `util`
- move `batchFileAdapter`, `defaultFilteredAdapter` and `fileAdapter` to `casbin/lib/esm/adapter`, and rename `DefaultFilteredAdapter` to `FilteredAdapter`
- `newEnforcer()` no longer supports the incoming file path and needs to pass model and adapter instances

Signed-off-by: Zixuan Liu <nodeces@gmail.com>

### CODE REVIEW

**Comment by nodece** [comment]
hi @harveyappleton and @Sefriol, could you review this pr?

**Review by Sefriol — COMMENTED** [review]
I think the changes are good, but I am just pondering if this is the right way to do it.

**Comment by hsluoyz** [comment]
@Zxilly

**Review by hsluoyz — CHANGES_REQUESTED** [review]
I don't think so many breaking changes are necessary. This will bring troubles to our existing Node-Casbin users. It's not worth doing it just for adapting to browser version Casbin.

File: README.md [hunk 1]
```diff
@@ -50,20 +52,62 @@ yarn add casbin
...
-```node.js
-// For Node.js:
+For Node.js:
+
+```typescript
+// For CommonJS:
 const { newEnforcer } = require('casbin');
-// For browser:
-// import { newEnforcer } from 'casbin';
+const { FileAdapter } = require('casbin/lib/cjs/adapter/node');
```
[the remark this answers is not in the document]
Comment (nodece) [reply README.md:60]: I think we should keep the `FileAdapter` in this package.

**Comment by hsluoyz** [comment]
Closed for now.

**Comment by Sefriol** [comment]
Yeah. Unless you want to release 6.0 version and keep 5.x updated for some time, there isn't really good way to do this.

**Pull request closed without merging**